adapters/sidekiq/lib/opentelemetry/adapters/sidekiq/middlewares/client/tracer_middleware.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sidekiq
+      module Middlewares
+        module Client
+          class TracerMiddleware
+            def call(_worker_class, job, _queue, _redis_pool)
+              tracer.in_span(
+                job['wrapped']&.to_s || job['class'],
+                attributes: {
+                  'job_id' => job['jid'],
```

Comment:
I'm not exactly sure what we should do about attributes that don't fall under a semantic convention. I think we should definitely record this, but I'm wondering if it should have a prefix of some kind. I'm also wondering if this is a common enough attribute that it deserves to have a convention defined. To be clear, I'm not exactly asking for anything here, just kind of thinking out loud on this.

Reply:
Should this be `messaging.message_id`? From https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md

Reply:
I think `messaging.message_id` fits here, going to use that.

Reply:
Of the already specified attributes, it's the closest match, but I'm not sure that it's semantically the same. I believe this field is supposed to represent the unique id assigned to a message as part of the underlying messaging protocol. Because background jobs involve operations such as consume, and produce, and listen on queues they intersect with messaging concepts, but aren't a perfect fit. For sidekiq, the underlying messaging system is Redis, which again, falls weirdly in between a datastore and a message broker. I think we can go with `messaging.message_id` for now, but perhaps have a discussion about conventions for background jobs more generally. I'll try to find out what other SIGs are doing for background jobs. There are similar systems, such as Celery for Python, for example.